### ISSUE TITLE
Make front-page loading spinner accessible

### ISSUE THREAD

**Issue by patcon** [GCTC-NTGC/gc-digital-talent#2573]
<sup>Originally suggested by Grant [on Slack](https://talent-cloud.slack.com/archives/CB424V61J/p1635521467031500?thread_ts=1635520995.031000&cid=CB424V61J) 🔒 </sup>

The loading spinner that refreshes on each change to the form is not accessible. What's needed (from first blush, likely `aria-live` and `aria-busy` attributes) is covered in this great tutorial :)

https://dockyard.com/blog/2020/03/02/accessible-loading-indicatorswith-no-extra-elements

This is likely a one-off for now, but perhaps whoever does the work could think on whether there are any patterns for this moving forward, because any loading content will need these affordances 🙏 (also, perhaps a little research, to not reinvent)

## Screenshot

<img width="346" alt="Screen Shot 2022-04-27 at 12 34 05 PM" src="https://user-images.githubusercontent.com/305339/165569580-bcb03ad6-d1e4-429d-bb4d-e0a1a5523c0f.png">

## Relevant Code

https://github.com/GCTC-NTGC/gc-digital-talent/blob/3a5ef34603b7ca8f3bd878c7a4273c0951a40299/frontend/talentsearch/src/js/components/search/EstimatedCandidates.tsx#L14-L20

---

Things like this make me want to lean on other frameworks and not creating our design system from the ground up, as per @esizer's comment on modals in Slack (esp when we're still learning). Other systems have already figured out some of this stuff (even though we would need to be constrained by their choices)

https://carbondesignsystem.com/patterns/loading-pattern/#accessibility

**Comment by esizer** [comment]
This is a good catch, I think we could using something like `role="status"` as well which would, by default, apply an `aria-live="polite"`. We could then control the busy indication depending on if the query is pending or not, something like `aria-busy={updatePending}`

Ref: https://developer.mozilla.org/en-US/docs/web/accessibility/aria/roles/status_role

In regards to announcing, assuming we don't already have a toast doing this, we could add some text to the spinner such as loading, searching, etc.

Though, if we do this, it may be valuable to replace the component rather than injecting the spinner in the message since it would produce some odd text: "There is approximately **searching** candidate right now who meets your criteria."

## Example

```tsx
...
 function weight(msg: string) { 
   return updatePending ? ( 
     <span className="lds-dual-ring" role="status" aria-busy="true"> 
        <span data-h2-visibility="b(invisible)">Searching&hellip;</span>
     </span>
   ) : ( 
     <span data-h2-font-weight="b(800)" role="status">{msg}</span> 
   ); 
 } 
...
```

**Comment by patcon** [comment]
>  I think we could using something like role="status" as well which would, by default, apply an aria-live="polite".

This sounds great! I'm learning.

But even though we get some default aria-live stuff under that role, wonder your thoughts on still explicitly stating it? I ask since we're still building this capacity in the team (myself included!), and a bit more legibility/expressiveness in the codebase feels better than jumping straight to "advanced" stuff that hides the more general practice. I share the above sensibility as one comparable to how I might not always use the most terse or advanced programming language features if some simpler and more verbose construct was easier to grok by the drive-by contributors I wish to engage.

Thoughts? Pls do be honest if that sounds silly or misguided, as I apply that logic in lots of places :)

**Comment by esizer** [comment]
> But even though we get some default aria-live stuff under that role, wonder your thoughts on still explicitly stating it? I ask since we're still building this capacity in the team (myself included!), and a bit more legibility/expressiveness in the codebase feels better than jumping straight to "advanced" stuff that hides the more general practice. I share the above sensibility as one comparable to how I might not always use the most terse or advanced programming language features if some simpler and more verbose construct was easier to grok by the drive-by contributors I wish to engage.
> 
> Thoughts? Pls do be honest if that sounds silly or misguided, as I apply that logic in lots of places :)

This makes sense, since `role="status"` isn't always appropriate it would be better to be explicit. I think  some thing along the lines of this could work:

```tsx
...
<div aria-live="polite" aria-busy={updatePending}>
  <p data-h2-text-align="b(center)">
    {updatePending ? (
      <>
        {intl.formatMessage({
          defaultMessage: "Searching...",
          description: "Message to display when a search is in progress.",
        })}
      </>
    ) : (
      <>
        {intl.formatMessage(
          {
            defaultMessage: `{candidateCount, plural, =1 {There is approximately <weight>{candidateCount} candidate right now who meets your criteria.} other {There are approximately <weight>{candidateCount}</weight> candidates right now who meet your criteria.}}`,
            description:
              "Message for total estimated candidates box next to search form.",
          },
          {
            candidateCount,
          }
        )}
      </>
    )}
  </p>
</div>
...
```

The `aria-live="polite"` will announce any changes to the assistive technology without interrupting the current action. Where as the `aria-busy={updatePending}` will notify the technology that the current region will update soon but is working on those updates not to not announce any changes during this. The `aria-busy` essentially prevents the region from being too chatty and sending.

## aria-busy

The `aria-busy` may be something we want to omit since it suggests that while it is set to `true` the assistive technology should not announce anything. So, it could be valuable to not include `aria-busy` if we always want the user to know a search is happening. 

I believe this is one of the more subjective areas for this, it really comes down to if we want the area to more or less noisy about what is occurring in it.

I would recommend we omit it in this case since I see the `aria-busy` being used when we expect multiple items to change at different intervals. In this case, we definitely wouldn't want the area to announce all changes as they happen and be patient and wait for them all to complete.

Im the scenario of the spinner, I think it is important to always tell the user when a search is in progress, especially if they are on a spotty connection and a request takes more than a few seconds to process.

### Examples

#### Set to False

1. Search is submitted
2. Announces "Searching"
3. Announces - X candidates found

#### Set to True

1. Search is submitted
2. 1-2 seconds pass
3. Announces - X Candidates found
